src/Common/Uuid/UuidRegistry.php

```diff
@@ -207,38 +207,45 @@ public function createMissingUuids()
         $counter = 0;
 
         // Empty should be NULL, but to be safe also checking for empty and null bytes
-        $resultSet = sqlStatementNoLog("SELECT * FROM `" . $this->table_name . "` WHERE `uuid` IS NULL OR `uuid` = '' OR `uuid` = '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0'");
-        while ($row = sqlFetchArray($resultSet)) {
-            if (!$this->table_vertical) {
-                // standard case, add missing uuid
-                sqlQueryNoLog("UPDATE `" . $this->table_name . "` SET `uuid` = ? WHERE `" . $this->table_id . "` = ?", [$this->createUuid(), $row[$this->table_id]]);
-                $counter++;
-            } else {
-                // more complicated case where setting uuid in a vertical table. In this case need a uuid for each combination of table columns stored in $this->table_vertical array
-                $stringQuery = "";
-                $arrayQuery = [];
-                $prependAnd = false;
-                foreach ($this->table_vertical as $column) {
-                    if ($prependAnd) {
-                        $stringQuery .= " AND `" . $column . "` = ? ";
+        $done = false;
+        while (!$done) {
+            // just maximum of 1000 at a time to attempt to speed things up and not break when inserting a large number of uuids
```

Comment:
plan to try 100, 500, 1000, 10000 to see what is best (or if there is even any difference)

Reply:
just gonna stick with 1000 for now. this is not really speeding things up (or slowing things down), but it definitely prevents breaking when doing very large updates (when having to do more than 500,000 uuid updates on a table).

Reply:
![addams_family](https://user-images.githubusercontent.com/5618372/110705332-3f452000-81c4-11eb-928b-4cf863c5983d.gif)